### PR TITLE
Fixed tests for transactions store

### PR DESCRIPTION
**Pull request by honza-kasik**

### CODE REVIEW

File: basic/src/test/java/org/jboss/hal/testsuite/test/configuration/transactions/JDBCStoreTestCase.java [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        JDBC_STORE_DATASOURCE_NAME = createDatasource();
+        operations.batch(new Batch()
+                .writeAttribute(TRANSACTIONS_ADDRESS, JDBC_STORE_DATASOURCE, JDBC_STORE_DATASOURCE_NAME)
+                .writeAttribute(TRANSACTIONS_ADDRESS, USE_JDBC_STORE, true)
+        ).assertSuccess();
+    }
+
+    @AfterClass
+    public static void afterClass() throws InterruptedException, TimeoutException, IOException, CommandFailedException {
+        client.apply(new RemoveDataSource(JDBC_STORE_DATASOURCE_NAME));
```
Comment (zebrik) [inline basic/src/test/java/org/jboss/hal/testsuite/test/configuration/transactions/JDBCStoreTestCase.java]: Removing datasource before restoring snapshot is not necessary IMO.

File: basic/src/test/java/org/jboss/hal/testsuite/test/configuration/transactions/JDBCStoreTestCase.java [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+        final String datasource = createDatasource();
+
+        try {
+            new ConfigChecker.Builder(client, TRANSACTIONS_ADDRESS)
+                    .configFragment(page.getConfigFragment())
+                    .editAndSave(ConfigChecker.InputType.TEXT, JDBC_STATE_STORE_TABLE_PREFIX, datasource)
+                    .verifyFormSaved()
+                    .verifyAttribute(JDBC_STATE_STORE_TABLE_PREFIX, datasource);
+        } finally {
+            client.apply(new RemoveDataSource(datasource));
```
Comment (zebrik) [inline basic/src/test/java/org/jboss/hal/testsuite/test/configuration/transactions/JDBCStoreTestCase.java]: I don't suggest removing datasource which is still used in configuration.

**Review by zebrik — CHANGES_REQUESTED** [review]
Is it possible to add tests for testing functionality of `Use journal store` and `Use jdbc store` checkboxes as well? I understand it requires additional restarts - necessary cleanup may be done inside these new test methods. 
These new tests may be added in new PR if you prefer it.

**Comment by honza-kasik** [comment]
Thanks! I added suggested tests (no restart needed after all) and fixed other suggestions too.

**Comment by zebrik** [comment]
Thank you! Merging...

**Pull request merged**